plugins/hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GDatabase.java

```diff
@@ -0,0 +1,281 @@
+package edu.berkeley.ground.plugins.hive;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.metastore.api.InvalidInputException;
+import org.apache.hadoop.hive.metastore.api.InvalidObjectException;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hive.common.util.HiveStringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+import edu.berkeley.ground.api.models.Edge;
+import edu.berkeley.ground.api.models.EdgeVersion;
+import edu.berkeley.ground.api.models.Node;
+import edu.berkeley.ground.api.models.NodeFactory;
+import edu.berkeley.ground.api.models.NodeVersion;
+import edu.berkeley.ground.api.models.NodeVersionFactory;
+import edu.berkeley.ground.api.models.Structure;
+import edu.berkeley.ground.api.models.StructureVersion;
+import edu.berkeley.ground.api.models.Tag;
+import edu.berkeley.ground.api.versions.GroundType;
+import edu.berkeley.ground.exceptions.GroundException;
+import edu.berkeley.ground.plugins.hive.GroundStore.EntityState;
+
+public class GDatabase {
```

Comment:
Oh, I see. Can we call them `GroundDatabase`, `GroundMetastore`, etc. to be more explicit?